Codec/DicomJpegLsCodec.cs

```diff
@@ -252,65 +253,82 @@ public override unsafe void Encode(DicomPixelData oldPixelData, DicomPixelData n
                     jls.allowedLossyError = jparams.AllowedError;
                 }
 
-                for (int frame = 0; frame < oldPixelData.NumberOfFrames; frame++)
-                {
-                    IByteBuffer frameData = oldPixelData.GetFrame(frame);
+                var pool = ArrayPool<byte>.Shared;
+                byte[] jpegData = null;
+                byte[] newJpegData = null;
 
-                    //Converting photmetricinterpretation YbrFull or YbrFull422 to RGB
-                    if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull)
-                    {
-                        frameData = PixelDataConverter.YbrFullToRgb(frameData);
-                    }
-                    else if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull422)
+                try
+                {
+                    for (int frame = 0; frame < oldPixelData.NumberOfFrames; frame++)
                     {
-                        frameData = PixelDataConverter.YbrFull422ToRgb(frameData, oldPixelData.Width);
-                    }
+                        IByteBuffer frameData = oldPixelData.GetFrame(frame);
 
-                    PinnedByteArray frameArray = new PinnedByteArray(frameData.Data);
+                        //Converting photmetricinterpretation YbrFull or YbrFull422 to RGB
+                        if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull)
+                        {
+                            frameData = PixelDataConverter.YbrFullToRgb(frameData);
+                        }
+                        else if (oldPixelData.PhotometricInterpretation == PhotometricInterpretation.YbrFull422)
+                        {
+                            frameData = PixelDataConverter.YbrFull422ToRgb(frameData, oldPixelData.Width);
+                        }
 
-                    byte[] jpegData = new byte[frameData.Size];
+                        PinnedByteArray frameArray = new PinnedByteArray(frameData.Data);
 
-                    try
-                    {
-                        fixed (byte* jpegDataPointer = jpegData)
+                        jpegData = pool.Rent((int)frameData.Size);
```

Comment:
Converting to int since there are no overloads for long while renting. Hope that is okay for frame. I can see below that we are resizing anyway to int.